### PR TITLE
added HPA handling and bugfixes with input deployment

### PR DESCRIPTION
**Pull request by rms1000watt**

### CODE REVIEW

File: k8s-canary-rollout.sh [hunk 1]
```diff
@@ -29,31 +28,37 @@ healthcheck(){
...
     kubectl apply -f $WORKING_VOLUME/original_deployment.yaml -n $NAMESPACE
     kubectl rollout status deployment/$PROD_DEPLOYMENT
 
     #we could also just scale to 0.
-    echo "[CANARY DELETE] Removing canary completely"
+    echo "[CANARY DELETE] Removing canary deployment completely"
     kubectl delete deployment $CANARY_DEPLOYMENT
 
+    echo "[CANARY DELETE] Removing canary horizontal pod autoscaler completely"
+    kubectl delete deployment $CANARY_DEPLOYMENT -n $NAMESPACE
```
Comment (kostis-codefresh) [inline k8s-canary-rollout.sh]: This doesn't look correct. It seems like the code just the deletes the deployment twice

File: k8s-canary-rollout.sh [hunk 2]
```diff
@@ -29,31 +28,37 @@ healthcheck(){
...
 
-
 cleanup(){
     echo "[CANARY CLEANUP] removing previous deployment $PROD_DEPLOYMENT"
     kubectl delete deployment $PROD_DEPLOYMENT -n $NAMESPACE
-    echo "[CANARY CLEANUP] marking canary as new production"
-    kubectl get service $SERVICE_NAME -o=yaml --namespace=${NAMESPACE} | sed -e "s/$CURRENT_VERSION/$NEW_VERSION/g" | kubectl apply --namespace=${NAMESPACE} -f - 
 
+    echo "[CANARY CLEANUP] removing previous horizontal pod autoscaler $PROD_DEPLOYMENT"
+    kubectl delete hpa $PROD_DEPLOYMENT -n $NAMESPACE
```
Comment (kostis-codefresh) [inline k8s-canary-rollout.sh]: Will this command succeed if the user of the canary plugin does NOT have HPA running? Maybe we need an global variable to decided if the HPA was present or not before the rolling update?

**Comment by kostis-codefresh** [comment]
Maybe I am missing something but I don't see
1) The parts where we backup HPA and restore it afterwards
2) Some conditional code to handle cases where HPA is not present at all.

File: k8s-canary-rollout.sh (hunk 1 above)
Comment (rms1000watt) [reply k8s-canary-rollout.sh]: Dohhh! let me fix

File: k8s-canary-rollout.sh (hunk 2 above)
Comment (rms1000watt) [reply k8s-canary-rollout.sh]: the command  will fail, but this bash  allows for exit codes > 0. There isn't  much error checking in the script in general. (like you can define a DEPLOYMENT that doesn't exist and the  script will chug along. so I'm just sticking to the current conventions)

**Comment by rms1000watt** [comment]
1. There isn't. This is for the usecase that someone has an INPUT_DEPLOYMENT with  an HPA defined. That usecase can be added later(?)
2. The code shouldn't fail if it's not present.

**Pull request merged**